homeassistant/components/agent_dvr/__init__.py

```diff
@@ -0,0 +1,129 @@
+"""Support for Agent."""
+import logging
+
+from agent.a import Agent
+
+from homeassistant.auth.permissions.const import POLICY_CONTROL
+from homeassistant.const import ATTR_ENTITY_ID, ENTITY_MATCH_ALL
+from homeassistant.exceptions import Unauthorized, UnknownUser
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.service import async_extract_entity_ids
+
+from .camera import CAMERA_SERVICES
+from .const import DEFAULT_RETRY, DOMAIN as AGENT_DOMAIN, SERVER_URL
+from .helpers import service_signal
+
+ATTRIBUTION = "ispyconnect.com"
+DEFAULT_BRAND = "Agent DVR by ispyconnect.com"
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_TIMEOUT = 10
+FORWARDS = ["camera"]
+
+
+async def async_setup(hass, config):
+    """Old way to set up integrations."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the Agent component."""
+
+    async def retry_setup(now):
+        """Retry setup if a connection/timeout happens on Agent API."""
+        await async_setup_entry(hass, config_entry)
+
+    hass.data.setdefault(AGENT_DOMAIN, {})
+
+    server_origin = config_entry.data[SERVER_URL]
+    if not server_origin:
+        return False
+    agent_client = Agent(server_origin)
+    await agent_client.update()
+    if not agent_client.is_available:
```

Comment:
Just throw a `ConfigEntryNotReady` exception here, Home Assistant will retry it automatically.